java/configuration/src/main/java/sleeper/configuration/table/index/DynamoDBTableIndex.java

```diff
@@ -86,30 +89,44 @@ public void create(TableStatus table) throws TableAlreadyExistsException {
                                 .withTableName(idIndexDynamoTableName)
                                 .withItem(idItem)
                                 .withConditionExpression("attribute_not_exists(#tableid)")
-                                .withExpressionAttributeNames(Map.of("#tableid", TABLE_ID_FIELD))));
+                                .withExpressionAttributeNames(Map.of("#tableid", TABLE_ID_FIELD))),
+                        new TransactWriteItem().withPut(new Put()
+                                .withTableName(onlineIndexDynamoTableName)
+                                .withItem(idItem)
+                                .withConditionExpression("attribute_not_exists(#tablename)")
+                                .withExpressionAttributeNames(Map.of("#tablename", TABLE_NAME_FIELD))));
         try {
             TransactWriteItemsResult result = dynamoDB.transactWriteItems(request);
             List<ConsumedCapacity> consumedCapacity = result.getConsumedCapacity();
             double totalCapacity = consumedCapacity.stream().mapToDouble(ConsumedCapacity::getCapacityUnits).sum();
             LOGGER.debug("Created table {}, capacity consumed = {}", table, totalCapacity);
         } catch (TransactionCanceledException e) {
-            CancellationReason nameIndexReason = e.getCancellationReasons().get(0);
-            if (isCheckFailed(nameIndexReason)) {
+            if (anyCheckFailed(e)) {
                 throw new TableAlreadyExistsException(table);
-            } else {
-                throw e;
             }
```

Comment:
This can keep the original exception as the cause of the TableAlreadyExistsException.